libtenzir/include/tenzir/select_optimization.hpp

```diff
@@ -0,0 +1,41 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <caf/default_sum_type_access.hpp>
+#include <caf/detail/type_list.hpp>
+#include <caf/none.hpp>
+#include <caf/variant.hpp>
+
+#include <string>
+#include <vector>
+
+namespace tenzir {
+
+class select_optimization {
+public:
+  std::vector<std::string> fields_of_interest{};
```

Comment:
```suggestion
struct select_optimization {
  std::vector<std::string> fields_of_interest{};
```

This expresses more clearly to the reader that all members are public intentionally.